node/src/arrow.ts

```diff
@@ -59,6 +59,14 @@ export async function convertToTable<T> (data: Array<Record<string, unknown>>, e
       if (typeof values[0] === 'string') {
         // `vectorFromArray` converts strings into dictionary vectors, forcing it back to a string column
         records[columnsKey] = vectorFromArray(values, new Utf8())
+      } else if (Array.isArray(values[0]) && typeof values[0][0] === 'string') {
```

Comment:
What if `values` contains a single empty array like `[[]]`?  You'll get an error trying `values[0][0]`

Reply:
i changed it to give a more helpful error message. Ideally we would pass in the schema here, or keep going to the next element til we get a non-empty observation otherwise.